data-prepper-plugins/kafka-plugins/src/integrationTest/java/org/opensearch/dataprepper/plugins/kafka/source/ConfluentKafkaProducerConsumerWithSchemaRegistryIT.java

```diff
@@ -0,0 +1,376 @@
+        when(avroTopicConfig.getMaxPollInterval()).thenReturn(Duration.ofSeconds(15));
+
+        encryptionConfig = mock(EncryptionConfig.class);
+
+        when(jsonSourceConfig.getAuthConfig()).thenReturn(authConfig);
+        when(jsonSourceConfig.getAcknowledgementsEnabled()).thenReturn(false);
+        when(jsonSourceConfig.getSchemaConfig()).thenReturn(schemaConfig);
+        when(jsonSourceConfig.getTopics()).thenReturn(List.of(jsonTopicConfig));
+        when(jsonSourceConfig.getBootstrapServers()).thenReturn(List.of(bootstrapServers));
+        when(jsonSourceConfig.getEncryptionConfig()).thenReturn(encryptionConfig);
+
+        when(avroSourceConfig.getAuthConfig()).thenReturn(authConfig);
+        when(avroSourceConfig.getAcknowledgementsEnabled()).thenReturn(false);
+        when(avroSourceConfig.getSchemaConfig()).thenReturn(schemaConfig);
+        when(avroSourceConfig.getTopics()).thenReturn(List.of(avroTopicConfig));
+        when(avroSourceConfig.getBootstrapServers()).thenReturn(List.of(bootstrapServers));
+        when(avroSourceConfig.getEncryptionConfig()).thenReturn(encryptionConfig);
+
+        when(encryptionConfig.getType()).thenReturn(EncryptionType.SSL);
+        when(plainTextAuthConfig.getUsername()).thenReturn(username);
+        when(plainTextAuthConfig.getPassword()).thenReturn(password);
+        when(schemaConfig.getBasicAuthCredentialsSource()).thenReturn("USER_INFO");
+        String[] userInfoParts = schemaRegistryUserInfo.split(":");
+        when(schemaConfig.getSchemaRegistryApiKey()).thenReturn(userInfoParts[0]);
+        when(schemaConfig.getSchemaRegistryApiSecret()).thenReturn(userInfoParts[1]);
+        when(schemaConfig.getRegistryURL()).thenReturn(schemaRegistryUrl);
+    }
+
+    @Test
+    public void KafkaJsonProducerConsumerTest() {
+        when(schemaConfig.getVersion()).thenReturn(2);
+        Thread producerThread = new Thread(() -> {
+            try {
+                produceJsonRecords(bootstrapServers,  numRecordsProduced);
+            } catch (Exception e){}
+        });
+        producerThread.setName("json-producer");
+        producerThread.start();
+        Thread consumerThread = new Thread(() -> {
+            try {
+                Thread.sleep(2000);
+                consumeRecords(bootstrapServers, jsonSourceConfig);
+            } catch (Exception e){}
+        });
+        consumerThread.setName("json-consumer");
+        consumerThread.start();
+        try {
+            producerThread.join();
+            consumerThread.join();
+        } catch (Exception e){}
+        assertThat(numRecordsReceived.get(), equalTo(numRecordsProduced));
+    }
+
+    public void consumeRecords(String servers, KafkaSourceConfig sourceConfig) {
+        kafkaSource = new KafkaSource(sourceConfig, pluginMetrics, acknowledgementSetManager, pipelineDescription);
+        kafkaSource.start(buffer);
+        for (int i = 0; i < 150; i++) {
```

Comment:
Similar to my comments above. I think you don't need any of this. Use Awaitility.